src/mrack/providers/provider.py

```diff
@@ -106,34 +97,97 @@ async def provision_hosts(self, hosts):
         logger.info(f"{self.dsp_name}: Provisioning duration: {provi_duration}")
 
         hosts = [self.to_host(srv) for srv in server_results]
-        errors = self.parse_errors(hosts)
+        error_hosts = self.parse_error_hosts(hosts)
+        success_hosts = [h for h in hosts if h not in error_hosts]
```

Comment:
Instead of several _list comprehension_ calls you can replace them with one `for` loop to fill those lists. This can be considered a nitpick because the number of hosts is usually not big.